File: src/org/lineageos/xiaomi_bluetooth/BootCompletedReceiver.java
package org.lineageos.xiaomi_bluetooth;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.UserHandle;
import android.util.Log;

public class BootCompletedReceiver extends BroadcastReceiver {

    private static final String TAG = BootCompletedReceiver.class.getSimpleName();
    private static final boolean DEBUG = true;

    @Override
    public void onReceive(Context context, Intent intent) {
        if (!Intent.ACTION_LOCKED_BOOT_COMPLETED.equals(intent.getAction())) {
            return;
        }
        if (DEBUG) Log.d(TAG, "boot completed");

        context.startServiceAsUser(
                new Intent(context, EarbudsService.class), UserHandle.CURRENT);
    }

}

File: src/org/lineageos/xiaomi_tws/BootCompletedReceiver.kt
package org.lineageos.xiaomi_tws

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import org.lineageos.xiaomi_tws.utils.ATUtils

class BootCompletedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_LOCKED_BOOT_COMPLETED != intent.action) {
            return
        }
        if (DEBUG) Log.d(TAG, "boot completed")

        ATUtils.checkIfSupportXiaomiATCommand()

        context.startService(Intent(context, EarbudsService::class.java))
    }

    companion object {
        private val TAG = BootCompletedReceiver::class.java.simpleName
        private const val DEBUG = true
    }
}
